### DISCUSSION TITLE
Kmm is is not build with swift package manager

### DISCUSSION THREAD

**Discussion by yegenpres** [codemagic-ci-cd/codemagic-docs#1987] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Kotlin Multiplatorm

### Steps to reproduce

  ios-kmm-workflow:
    name: Dansk assistent IOS KMM Workflow
    instance_type: mac_mini_m1
    max_build_duration: 120
    integrations:
      app_store_connect: Codemagic
    triggering:
      events:
        - push
      branch_patterns:
        - pattern: "release"
          include: true
    environment:
      java: 17
      ios_signing:
        distribution_type: app_store
        bundle_identifier: com.danskassistent.app
      vars:
        XCODE_PROJECT: "Dansk\ Assistent.xcodeproj"
        XCODE_SCHEME: "Dansk\ Assistent"
        APP_ID: 6449679395 # <-- Use the TestFlight Apple id number (An automatically generated ID assigned to your app) found under General > App Information > Apple ID.
      xcode: latest
    scripts:
      - name: Install Swift Packages
        script: |
          cd iosAssistentApp
          xcodebuild -resolvePackageDependencies
      - name: Build shared
        script: |
          cd iosAssistentApp
          /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -project Dansk\ Assistent.xcodeproj -scheme Dansk\ Assistent -configuration Debug OBJROOT=../build/ios SYMROOT=../build/ios -sdk iphonesimulator -arch arm64 -allowProvisioningDeviceRegistration -allowProvisioningUpdates
      - name: Set up code signing settings on Xcode project
        script: |
          cd $CM_BUILD_DIR/iosAssistentApp
          xcode-project use-profiles
      - name: Increment build number
        script: |
          #!/bin/sh
          set -e
          set -x
          cd $CM_BUILD_DIR/iosAssistentApp
          LATEST_BUILD_NUMBER=$(app-store-connect get-latest-app-store-build-number "$APP_ID")
          agvtool new-version -all $(($LATEST_BUILD_NUMBER + 5))
      - name: Build ipa for distribution
        script: |
          cd $CM_BUILD_DIR/iosAssistentApp
          xcode-project build-ipa \
            --project "$XCODE_PROJECT" \
            --scheme "$XCODE_SCHEME"
    artifacts:
      - iosAssistentApp/build/ios/ipa/*.ipa
    publishing:
      email:
        recipients:
          - jksemenyuk@gmail.com
        notify:
          success: true
          failure: true
      app_store_connect:
        auth: integration

        # Configuration related to TestFlight (optional)
        # Note: This action is performed during post-processing.
        submit_to_testflight: true
        beta_groups: # Specify the names of beta tester groups that will get access to the build once it has passed beta review.
          - family

        # Configuration related to App Store (optional)
        # Note: This action is performed during post-processing.
        submit_to_app_store: true

### Expected results

I expect to get build folder of shared module when run script "Build shared"
But I received next one: The following build commands failed:
	Copy /Users/builder/clone/build/ios/Debug-iphonesimulator/Dansk\ Assistent.app/RevenueCat_RevenueCat.bundle /Users/builder/clone/build/ios/Debug-iphonesimulator/RevenueCat_RevenueCat.bundle (in target 'Dansk Assistent' from project 'Dansk Assistent')
(1 failure)

It is not package problem, it happened with eany other package if I change it.

### Actual results

	Copy /Users/builder/clone/build/ios/Debug-iphonesimulator/Dansk\ Assistent.app/RevenueCat_RevenueCat.bundle /Users/builder/clone/build/ios/Debug-iphonesimulator/RevenueCat_RevenueCat.bundle (in target 'Dansk Assistent' from project 'Dansk Assistent')
(1 failure)

### Build id (optional)

64e3ab9dfa11e09d3013b666

**Comment by navasiloy** [comment]
I'm getting same issue 
it looks like it's failing when running 

`/usr/bin/codesign --force --sign 17CBB29F0C13E49C66251DFB205269AC3F6AD6E9 --generate-entitlement-der /Users/builder.../RevenueCat_RevenueCat.bundle` with `errSecInternalComponent`

build id 64e4fbc914e6077715765fb8

**Comment by kulakarla** [comment]
Hello @yegenpres 

From the errors, I managed to find this snippet:
`error: /Users/builder/clone/build/ios/Debug-iphonesimulator/GoogleSignIn_GoogleSignIn.bundle: No such file or directory (in target 'Dansk Assistent' from project 'Dansk Assistent')`

Can you please make sure your script with file paths is correct. You can also inspect the files in our VM-s by [enabling SSH/VNC access and connecting to our machines.](https://docs.codemagic.io/troubleshooting/accessing-builder-machine-via-ssh/)

**Comment by kulakarla** [comment]
@navasiloy @yegenpres 

I will have to reproduce this issue on my end, which may take some time.
If any of you have found the issue and its solution, please share it in the thread so we could consider it answered.

**Comment by himesh-cm** [comment]
Hi @yegenpres,
The build id `64e3ab9dfa11e09d3013b666`
is missing the step 
```
- name: Install Swift Packages
script: |
cd iosAssistentApp
xcodebuild -resolvePackageDependencies
```

Can you share the revised build id?

**Reply by yegenpres** [reply]
64e3ab9dfa11e09d3013b666 if was the same with this build, and it contain this command

**Reply by himesh-cm** [reply]
I cannot see it, after `Set up code signing identities` the next step is `Build shared`.
Can you share the screenshot or guide in which step you have added it?

**Reply by yegenpres** [reply]
which step is messing?

**Reply by himesh-cm** [reply]
Hi @yegenpres, Sorry for the delayed response. Your message was missed. I am investigating and will reply shortly.

**Reply by himesh-cm** [reply]
Hi @yegenpres & @navasiloy, I cannot reproduce this at my end. Can you create a minimum reproducible setup and share it via a public GitHub link?

Also, the issue seems to be with `RevenueCat_RevenueCat`. I tired looking at their forums for similar issues, Please check if this GH issue [helps](https://github.com/RevenueCat/purchases-ios/issues/3117).

Alternatively if you can run `verbose` build, i can further deep dive into the issue.

**Comment by himesh-cm** [comment]
Hi @yegenpres & @navasiloy, 

I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
